pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -237,6 +237,7 @@ func dropDisabledStatusFields(newClaim, oldClaim *resource.ResourceClaim) {
 	dropDisabledDRAResourceClaimDeviceStatusFields(newClaim, oldClaim)
 	dropDisabledDRAAdminAccessStatusFields(newClaim, oldClaim)
 	dropDisabledDRAResourceClaimConsumableCapacityStatusFields(newClaim, oldClaim)
+	dropDeviceBindingConditionsFields(newClaim, oldClaim)
```

Comment:
I am wondering whether this was an oversight or intentionally left out. Suppose a device has binding conditions and the feature gate is on in the scheduler while it is off in the apiserver (not a valid combination, but who knows...). The scheduler will then try to publish the allocation with binding conditions.

Should that succeed or should the fields get dropped?

Consistent with normal behavior would be  to drop them. It's not a good failure mode and the client cannot even protect against it, not even by asking for strict validation because supported, disabled fields get dropped silently.

I think we can treat this as an oversight and merge this PR.

cc @KobayashiD27